### PR TITLE
Fix benchmark reporting with subtitles of result key

### PR DESCRIPTION
**Pull request by afsalthaj**

### CODE REVIEW

**Comment by afsalthaj** [comment]
I have to merge this reporting changes for me to test a proper benchmark reporting. Unfortunately GitHub Actions doesn't allow me to run this action without merging.

File: integration-tests/src/benchmarks/report/benchmark_report.rs [hunk 1]
```diff
@@ -233,31 +250,35 @@ mod internal {
...
+        pub run_config: RunConfig,
+        pub result_key: ResultKey,
+    }
 
-            if length != 0 {
-                let avg = total_duration.div_f64(length as f64);
-                run_config_to_avg_time.insert(run_config.clone(), avg);
-            } else {
-                run_config_to_avg_time.insert(run_config.clone(), Duration::ZERO);
+    pub fn calculate_mean_avg_time(json: &BenchmarkResult) -> HashMap<ReportKey, Duration> {
```
Comment (afsalthaj) [inline integration-tests/src/benchmarks/report/benchmark_report.rs:259]: Change this name

**Review by justcoon — APPROVED** [review]

**Pull request merged**